### DISCUSSION TITLE
'@babel/register' not caching plugins and presets correctly

### DISCUSSION THREAD

**Discussion by pbadenski** [babel/babel#11650] [Q&A] [answered]
Dear community! :D

I was looking at long test times in our project. I recorded a CPU profile of a test run and noticed something rather odd. `loadPlugin` and `loadPreset` take over 40% of the test run and 10 seconds in total! See below:

<img width="1136" alt="Screenshot 2020-05-30 at 13 38 43" src="https://user-images.githubusercontent.com/68713/83328440-ffd3c580-a27a-11ea-89e2-02757fa0db64.png">

I then ran in a debugger to see what's going on in babel core and noticed that plugins and presets somehow get reloaded all the time whereas they should be loaded from cache.

To clarify - I double checked that there is multiple calls to `loadPlugin` and `loadPreset` and they're all roughly the same time. (rather than a single slow call and many very fast ones)

Below `babel-register.js` we use for mocha:
```
const options = {
  extensions: [ ".ts", ".tsx", ".js" ],
  presets: [
    "@babel/preset-react",
    "@babel/preset-typescript",
  ],
  plugins: [
    "@babel/plugin-transform-modules-commonjs",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-export-namespace-from",
  ],
  ignore: [ /node_modules\/(?!d3)/ ],
  babelrc: false,
  cache: true
};
require("@babel/register")(options);
````

Appreciate your help!

**Comment by nicolo-ribaudo** [comment, marked as answer]
The problem is that we still need to resolve the plugins every time. If you use `require("plugin name")` instead of just `"plugin name"` in your config, you might get a good perf boost because you completely skip Babel's resolution.

**Comment by pbadenski** [comment]
Wow, thanks, this is a great tip! This reduced our test startup time by 50-70% on average! Wish I'd knew about it earlier :D
